README.md

```diff
@@ -1,48 +1,455 @@
+`in Template`
 
+```html
+<ht-table [data]="this.datasource" [columnConfigs]="this.columnConfigs"> </ht-table>
 ```
+
+`in Component`
+
+```ts
+public datasource?: TableDataSource<TableRow> = {
+  getData : () => of({
+    data: [{name: 'test-name1'}, {name: 'test-name2'}],
+    totalCount: 2
+  })
+};
+public readonly columnConfigs: TableColumnConfig[] = [
+  {
+    id: 'name',
+    title: 'Name'
+    visible: true,
+    sortable: false,
+    width: '48px',
+  }
+];
+```
+
+Now let's break this down.
+
+- It will create a table with a single column (In column configs we have only mentioned one column) and two rows (in the data source we have mentioned data as array of two objects.).
+- Now if we look closely, table column config's id is same as the key we have used the key in data which is `name`. This is must for the table to render the data correctly.
+
+There is a lot there in the tables, like custom controls, configurations (for pagination and many other). We highly recommend you to check out the `table.component.ts` to learn about tables and check out all the different examples present in the application.
+
+#### Cell Renderers
+
+As we have talked about tables, now let's talk about the custom table cell renderers. in hypertrace-ui, we can create a custom table cell renderer to display the data in specifc format. These are nothing but angular component with another decorator `TableCellRenderer`
+<br />
+Now let's see how we can create a custom cell renderer. for example `hello-table-cell-renderer.component.ts`
+
+```ts
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { TableCellRenderer } from '../../table-cell-renderer';
+import { TableCellRendererBase } from '../../table-cell-renderer-base';
+import { CoreTableCellParserType } from '../../types/core-table-cell-parser-type';
+import { TableCellAlignmentType } from '../../types/table-cell-alignment-type';
+
+@Component({
+  selector: 'ht-hello-table-cell-renderer',
+  styleUrls: ['./hello-table-cell-renderer.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: ` <div>Hello {{ this.value }}</div> `
+})
+@TableCellRenderer({
+  type: 'hello',
+  alignment: TableCellAlignmentType.Left,
+  parser: CoreTableCellParserType.NoOp
+})
+export class HelloTableCellRendererComponent extends TableCellRendererBase<string> {}
+```
+
+Now let's break this down.
+
+- We have used decorator `TableCellRenderer` by which we're registering this cell renderer (on build) for usage.
+- `type` property is the unique string to define each cell renderer.
+- `alignment` is used for the cell alignment, could be - left, right and center
+- `parser` is used to parse the data, it is also defined in the similar way. Suppose we are getting a data as `['test1', 'test2']` and we want it be used like `{value1: 'test1', value2: 'test2'}` then we can create a custom parser and can handle this use case. For no operation we use `CoreTableCellParserType.NoOp`
```

Comment:
```suggestion
- `parser` is used to parse the data. It can also be defined similar to a cell renderer. Suppose we are getting data as `['test1', 'test2']` and we want it to be marshalled into `{value1: 'test1', value2: 'test2'}` then we can create a custom parser and can handle the transformation. For no operation we use `CoreTableCellParserType.NoOp`
```